addons/binding/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/UniFiController.java

```diff
@@ -0,0 +1,403 @@
+            logger.trace("Sending the following headers to {}", url);
+            Map<String, List<String>> headers = connection.getRequestProperties();
+            for (Map.Entry<String, List<String>> header : headers.entrySet()) {
+                logger.trace("{} : {}", header.getKey(), header.getValue());
+            }
+        }
+        return connection;
+    }
+
+    private String get(String url, Map<String, String> params, boolean loginGuard)
+            throws IOException, URISyntaxException {
+        String response = null;
+        HttpsURLConnection connection = openConnection(url);
+        if (params != null) {
+            connection.setDoOutput(true);
+            objectMapper.writeValue(connection.getOutputStream(), params);
+        }
+        int rc = connection.getResponseCode();
+        if (rc == HTTP_OK) {
+            if (logger.isTraceEnabled()) {
+                logger.trace("Received the following headers from {}", url);
+                Map<String, List<String>> headers = connection.getHeaderFields();
+                for (Map.Entry<String, List<String>> header : headers.entrySet()) {
+                    logger.trace("{} : {}", header.getKey(), header.getValue());
+                }
+            }
+            cookieManager.put(connection.getURL().toURI(), connection.getHeaderFields());
+            response = IOUtils.toString(connection.getInputStream());
+        } else if (rc == HTTP_UNAUTHORIZED && !loginGuard) {
+            // our session has timed-out, let's try again
+            if (login()) {
+                return get(url, params, true); // guard against a "login loop"
+            }
+        }
+        return response;
+    }
+
+    private String get(String url, Map<String, String> params) throws IOException, URISyntaxException {
+        return get(url, params, false);
+    }
+
+    private String get(String url) throws IOException, URISyntaxException {
+        return get(url, (Map<String, String>) null);
+    }
+
+    private <T> T get(String url, Map<String, String> params, Class<T> responseType) throws UniFiException {
+        T response = null;
+        try {
+            String content = get(url, params);
+            if (StringUtils.isBlank(content)) {
+                logger.warn("Empty GET response from URL: {}", url);
+                throw new UniFiException("Could not communicate with the UniFi Controller");
+            } else {
+                if (logger.isTraceEnabled()) {
+                    logger.trace("JSON content from URL: {}\n{}", url, objectMapper.writerWithDefaultPrettyPrinter()
+                            .writeValueAsString(objectMapper.readValue(content, Object.class)));
+                }
+                response = objectMapper.readValue(content, responseType);
+            }
+        } catch (SocketException | SocketTimeoutException e) {
+            throw new UniFiException("Could not communicate with the UniFi Controller");
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception